hastory-data/src/Data/Hastory/Types/Schema.hs

```diff
@@ -0,0 +1,56 @@
+{-# LANGUAGE DeriveGeneric #-}
+{-# LANGUAGE GADTs #-}
+{-# LANGUAGE GeneralizedNewtypeDeriving #-}
+{-# LANGUAGE MultiParamTypeClasses #-}
+{-# LANGUAGE OverloadedStrings #-}
+{-# LANGUAGE QuasiQuotes #-}
+{-# LANGUAGE RecordWildCards #-}
+{-# LANGUAGE TemplateHaskell #-}
+{-# LANGUAGE TypeFamilies #-}
+
+module Data.Hastory.Types.Schema where
+
+import Data.Hastory.Types.Path ()
+
+import Control.DeepSeq
+import Data.Aeson
+import Data.Hashable
+import Data.Hashable.Time ()
+import Data.Text (Text)
+import Data.Time
+import Database.Persist.TH (mkMigrate, mkPersist, persistLowerCase, share, sqlSettings)
+import GHC.Generics (Generic)
+import Path
+
+-- Client Schema
```

Comment:
You don't want to put the client schema and the server schema in the same place.
You should put them in separate modules so that you can use the same name for each and use the functions qualified.
Ideally you'd put them in separate packages even:
`hastory-server-data` and `hastory-cli-data`.
The reason is that the name clashes actually help to prevent that someone uses the wrong database schema in their program.
Does that make sense?

Reply:
Yup - it does. I originally thought that putting them schemas in different packages would make sense, but given that I am still figuring out what idiomatic Haskell is, I didn't want to go down that route without confirmation. I'll make that change.

Reply:
Super, you can look at https://gitlab.com/Norfair/donna/-/tree/master/ for an example.

Reply:
I didn't see your comment in time, but I created two new packages:

- hastory-data-cli
- hastory-data-server
- hastory-path

And the dependency graph looks like this:

hastory-data  --> hastory-data-cli  --> hastory-path
                      --> hastory-data-server --> hastory-path

Unfortunately, it looks like I broke nix, since `nix-build` will not work on my local machine. I am trying to Google for a fix but I am not familiar at all with Nix.